modules/clients/src/test/java/org/apache/ignite/internal/processors/rest/JettyRestProcessorSecurityCreateDestroyPermissionTest.java

```diff
@@ -0,0 +1,306 @@
+    /** */
+    private static final String UNALLOWED_CLNT_SYSTEM = "unallowed_clnt_system";
+
+    /** */
+    protected static final ObjectMapper JSON_MAPPER = new GridJettyObjectMapper();
+
+    /** */
+    private static final String STATUS_FLD = "successStatus";
+
+    /** */
+    private static final String ERROR_FLD = "error";
+
+    /** */
+    private static final String SUCCESS_STATUS = "0";
+
+    /** */
+    private static final String ERROR_STATUS = "3";
+
+    /** */
+    private static final String CACHE_NAME = "TEST_CACHE";
+
+    /** */
+    private static final String UNDECLARED_CACHE = "UNDECLARED_CACHE";
+
+    /** */
+    private static final SecurityPermission[] EMPTY_PERM = new SecurityPermission[0];
+
+    /** */
+    private static final int DFLT_REST_PORT = 8091;
+
+    /** */
+    private static final String REST_URL ="http://" + LOC_HOST +":" + DFLT_REST_PORT + "/ignite?";
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        TestSecurityData[] clientData = new TestSecurityData[] {
+            new TestSecurityData(CLNT_ADMIN,
+                SecurityPermissionSetBuilder.create().defaultAllowAll(false)
+                    .appendSystemPermissions(ADMIN_CACHE)
+                    .build()
+            ),
+            new TestSecurityData(ALLOWED_CLNT_CACHE,
+                SecurityPermissionSetBuilder.create().defaultAllowAll(false)
+                    .appendCachePermissions(CACHE_NAME, CACHE_CREATE, CACHE_DESTROY)
+                    .build()
+            ),
+            new TestSecurityData(UNALLOWED_CLNT_CACHE,
+                SecurityPermissionSetBuilder.create().defaultAllowAll(false)
+                    .appendCachePermissions(CACHE_NAME, EMPTY_PERM)
+                    .build()
+            ),
+            new TestSecurityData(ALLOWED_CLNT_SYSTEM,
+                SecurityPermissionSetBuilder.create().defaultAllowAll(false)
+                    .appendSystemPermissions(CACHE_CREATE, CACHE_DESTROY)
+                    .build()
+            ),
+            new TestSecurityData(UNALLOWED_CLNT_SYSTEM,
+                SecurityPermissionSetBuilder.create().defaultAllowAll(false)
+                    .build()
+            )
+        };
+
+        return super.getConfiguration(igniteInstanceName)
+            .setPluginProviders(new TestSecurityPluginProvider(igniteInstanceName, "", ALLOW_ALL, false, clientData));
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        System.setProperty(IGNITE_JETTY_PORT, String.valueOf(DFLT_REST_PORT));
+
+        super.beforeTestsStarted();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected int gridCount() {
+        return 1;
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTest() throws Exception {
+        IgniteEx server = grid(0);
+
+        server.cacheNames().forEach(server::destroyCache);
```

Comment:
If you want to clear all, use stopAllGrids(); and cleanPersistenceDir();

Reply:
stopAllGrids and startGrid expensive operations